tests/Unit/Elliptic/BoundaryConditions/Test_BoundaryCondition.cpp

```diff
@@ -0,0 +1,176 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/BoundaryConditions/BoundaryCondition.hpp"
+#include "Domain/InterfaceHelpers.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Elliptic/Utilities/ApplyAt.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace elliptic::BoundaryConditions {
+
+namespace {
+
+struct ArgumentTag : db::SimpleTag {
+  using type = int;
+};
+
+struct NonlinearArgumentTag : db::SimpleTag {
+  using type = int;
+};
+
+struct VolumeArgumentTag : db::SimpleTag {
+  using type = bool;
+};
+
+template <typename Registrars>
+class TestBoundaryCondition;
+
+struct TestRegistrar {
+  template <typename Registrars>
+  using f = TestBoundaryCondition<Registrars>;
+};
+
+template <typename Registrars = tmpl::list<TestRegistrar>>
+class TestBoundaryCondition : public BoundaryCondition<1, Registrars> {
+ private:
+  using Base = BoundaryCondition<1, Registrars>;
+
+ public:
+  using options = tmpl::list<>;
+  static constexpr Options::String help = "halp";
+
+  TestBoundaryCondition() = default;
+  TestBoundaryCondition(const TestBoundaryCondition&) = default;
+  TestBoundaryCondition(TestBoundaryCondition&&) = default;
+  TestBoundaryCondition& operator=(const TestBoundaryCondition&) = default;
+  TestBoundaryCondition& operator=(TestBoundaryCondition&&) = default;
+  ~TestBoundaryCondition() override = default;
+  explicit TestBoundaryCondition(CkMigrateMessage* m) noexcept : Base(m) {}
+  WRAPPED_PUPable_decl_base_template(
+      SINGLE_ARG(BoundaryCondition<1, Registrars>),
+      SINGLE_ARG(TestBoundaryCondition<Registrars>));  // NOLINT
+
+  std::unique_ptr<domain::BoundaryConditions::BoundaryCondition> get_clone()
+      const noexcept override {
+    return std::make_unique<TestBoundaryCondition>(*this);
+  }
+
+  using argument_tags =
+      tmpl::list<ArgumentTag, VolumeArgumentTag, NonlinearArgumentTag>;
+  using volume_tags = tmpl::list<VolumeArgumentTag>;
+
+  /// [example_poisson_fields]
+  static void apply(
+      const int arg0, const bool arg1, const int arg2,
```

Comment:
maybe name these `argument_on_face`, `argument_from_volume`, `nonlinear_argument_on_face` or something like that. Since this is used in a snippet I think it would be good to make the names a bit clearer